packages/@headlessui-react/src/components/dialog/dialog.tsx

```diff
@@ -390,8 +390,7 @@ function DialogFn<TTag extends ElementType = typeof DEFAULT_DIALOG_TAG>(
     )
   }
 
-  let inTransitionComponent = usesOpenClosedState !== null
-  if (!inTransitionComponent && open !== undefined && !rest.static) {
```

Comment:
Now we are not relying on the `OpenClosed` state from a parent to know whether we should wrap the `Dialog` in a `Transition`

Instead, we will always do that. If you happen to already wrap your `Dialog` in a `Transition` on the outside, then the props will be forwarded to the underlying element because `Transition` renders as `Fragment` by default.